torch_geometric/data/graph_store.py

```diff
@@ -0,0 +1,126 @@
+from abc import abstractmethod
+from collections.abc import MutableMapping
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Dict, Optional, Union
+
+from torch import LongTensor
+
+from torch_geometric.typing import EdgeTensorType, EdgeType
+from torch_geometric.utils.mixin import CastMixin
+
+
+class EdgeLayout(Enum):
+    COO = 'coo'
+    CSC = 'csc'
+    CSR = 'csr'
+
+
+@dataclass
+class EdgeAttr(CastMixin):
+    r"""Defines the attributes of an :obj:`GraphStore` edge."""
+
+    # The type of the edge
+    edge_type: Optional[Any]
+
+    # The layout of the edge representation
+    layout: EdgeLayout
+
+    # Whether the edge index is sorted
+    is_sorted: bool = False
```

Comment:
Can we define what that means? Ideally by destination node. Also mention that it can only be set for `COO` format?